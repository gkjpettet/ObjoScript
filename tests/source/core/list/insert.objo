var list = ["a", "b", "c"]
System.print(list.toString) // expect [a, b, c]

list.insert(0, "d")
System.print(list.toString) // expect [d, a, b, c]
list = ["a", "b", "c"]

list.insert(1, "d")
System.print(list.toString) // expect [a, d, b, c]
list = ["a", "b", "c"]

list.insert(2, "d")
System.print(list.toString) // expect [a, b, d, c]
list = ["a", "b", "c"]

list.insert(3, "d")
System.print(list.toString) // expect [a, b, c, d]
list = ["a", "b", "c"]

list.insert(-1, "d")
System.print(list.toString) // expect [a, b, c, d]
list = ["a", "b", "c"]

list.insert(-2, "d")
System.print(list.toString) // expect [a, b, d, c]
list = ["a", "b", "c"]

list.insert(-3, "d")
System.print(list.toString) // expect [a, d, b, c]
list = ["a", "b", "c"]

list.insert(-4, "d")
System.print(list.toString) // expect [d, a, b, c]
list = ["a", "b", "c"]