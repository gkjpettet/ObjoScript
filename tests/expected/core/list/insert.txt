[a, b, c]
[d, a, b, c]
[a, d, b, c]
[a, b, d, c]
[a, b, c, d]
[a, b, c, d]
[a, b, d, c]
[a, d, b, c]
[d, a, b, c]